kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetSchemaUtils.java

```diff
@@ -265,38 +272,60 @@ private static Type toParquetType(
         return type;
     }
 
-    private static Type toParquetArrayType(ArrayType arrayType, String name, Repetition rep) {
-        // We will be supporting the 3-level array structure only. 2-level array structure will
-        // be supported in the future.
+    private static Type toParquetArrayType(
+            StructField nearestAncestor, // nearest ancestor with a StructField
+            String relativePath, // relative path to this element from the nearestAncestor
+            ArrayType arrayType,
+            String name,
+            Repetition rep) {
+        // We will be supporting the 3-level array structure only. 2-level array structures are
+        // a very old legacy versions of Parquet which Kernel doesn't support writing as.
+
+        String elementRelativePath = relativePath + ".element";
         return Types
                 .buildGroup(rep)
                 .as(LogicalTypeAnnotation.listType())
                 .addField(Types.repeatedGroup()
                         .addField(toParquetType(
+                                nearestAncestor,
+                                elementRelativePath,
                                 arrayType.getElementType(),
                                 "element", /* name */
                                 arrayType.containsNull() ? OPTIONAL : REQUIRED,
-                                Optional.empty()))
+                                getNestedFieldId(nearestAncestor, elementRelativePath)))
                         .named("list"))
                 .named(name);
     }
 
-    private static Type toParquetMapType(MapType mapType, String name, Repetition repetition) {
-        // We will be supporting the 3-level array structure only. 2-level array structure will
-        // be supported in the future.
+    private static Type toParquetMapType(
+            StructField nearestAncestor, // nearest ancestor with a StructField
```

Comment:
IMO params like this are worth defining with proper javadocs and an example. For example, can a StructField be the nearest ancestor of itself?